watertap/unit_models/tests/test_steam_heater_0D.py

```diff
@@ -0,0 +1,80 @@
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+)
+
+from idaes.core import FlowsheetBlock
+import watertap.property_models.water_prop_pack as props_w
+import watertap.property_models.seawater_prop_pack as props_sw
+from watertap.unit_models.steam_heater_0D import SteamHeater0D
+from idaes.core.solvers import get_solver
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+from idaes.models.unit_models.heat_exchanger import (
+    HeatExchangerFlowPattern,
+)
+import idaes.core.util.scaling as iscale
+from watertap.unit_models.tests.unit_test_harness import UnitTestHarness
+
+# Set up solver
+solver = get_solver()
+
+
+@pytest.mark.requires_idaes_solver
+@pytest.mark.component
+def build():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.steam_properties = props_w.WaterParameterBlock()
+    m.fs.cold_side_properties = props_sw.SeawaterParameterBlock()
+
+    m.fs.unit = SteamHeater0D(
+        hot_side_name="hot",
+        cold_side_name="cold",
+        hot={
+            "property_package": m.fs.steam_properties,
+        },
+        cold={
+            "property_package": m.fs.cold_side_properties,
+        },
+        delta_temperature_callback=delta_temperature_chen_callback,
+        flow_pattern=HeatExchangerFlowPattern.countercurrent,
+    )
+
+    m.fs.unit.hot_side_inlet.flow_mass_phase_comp[0, "Vap", "H2O"].set_value(6)
+    m.fs.unit.hot_side_inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(0)
+    m.fs.unit.hot_side_inlet.temperature.fix(273.15 + 140)
+    m.fs.unit.hot_side_inlet.pressure[0].set_value(201325)
+    m.fs.unit.cold_side_inlet.pressure.fix(101325)
+    m.fs.unit.cold_side_inlet.temperature.fix(273.15 + 70)
+    m.fs.unit.cold_side_inlet.flow_mass_phase_comp[0, "Liq", "TDS"].fix(0.035 * 6)
+    m.fs.unit.cold_side_inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(6)
+    m.fs.unit.area.fix(50)
+    m.fs.unit.overall_heat_transfer_coefficient.fix(2e3)
+
+    iscale.calculate_scaling_factors(m.fs.unit)
+
+    return m
+
+
+class TestSteamHeater0D(UnitTestHarness):
+    def configure(self):
+        m = build()
+
+        self.unit_solutions[
+            m.fs.unit.hot_side_inlet.flow_mass_phase_comp[0, "Vap", "H2O"]
+        ] = 2.92784431885
```

Comment:
I'm not very familiar with the guidance in WaterTAP to create tests. I'm going to follow the rules in IDAES temporarily. It seems too simple to me; at least, testing is needed for building blocks. Also, the most important saturation condition is not tested.

Reply:
Hi @Morgan88888888 - I believe @ElmiraShamlou is using a unit test harness, which actually runs a whole sequence of standard tests (e.g., testing the build). The nice part for developers familiar with it is that you don't have to write all tests from scratch, and you are less prone to forgetting one of the "must-have" tests. On the other hand, for those unfamiliar with it, it can be a little unclear what is happening. Here is a how-to on unit test harness: https://watertap.readthedocs.io/en/latest/how_to_guides/how_to_use_unit_test_harness.html

To see what tests run under the hood, you can also check out `watertap\unit_models\tests\unit_test_harness.py`

Reply:
@Morgan88888888 I echo Adam's points and would like to add that I am also integrating few more unit_solutions into this. I will include these updates in my next commit to this PR which includes the flowsheet integration.